cmake/util.cmake

```diff
@@ -0,0 +1,84 @@
+# Finds all transitive static library dependencies of a given target
```

Comment:
This is very similar to a utility in Shaderc.  

https://github.com/google/shaderc/blame/main/cmake/utils.cmake#L140

Reply:
Yes these are copied from shaderc. Do you want to mention where these are coming?

Reply:
No, its fine.  I looked because I think I reviewed the original thing in Shaderc, and it also gave me comfort that the logical was probably correct.